conf/app.routes

```diff
@@ -1,3 +1,4 @@
 # microservice specific routes
 
-DELETE      /:notificationId        @uk.gov.hmrc.apinotificationpull.controllers.NotificationsController.delete(notificationId)
+DELETE    /notification/:notificationId    @uk.gov.hmrc.apinotificationpull.controllers.NotificationsController.delete(notificationId)
+GET       /notifications/:clientId         @uk.gov.hmrc.apinotificationpull.controllers.NotificationsController.getAllByClientId(clientId)
```

Comment:
ClientId is a header passed as x-client-id not as a path parameter. Also this should be on the / route as it has a context of notifications already.